target-app/Onboarding/SignIn/Views/SignInViewController.swift

```diff
@@ -0,0 +1,154 @@
+//
+
+import UIKit
+
+class SignInViewController: UIViewController {
+    
+    private lazy var overlayImageView = UIImageView()
+    
+    private lazy var titleLabel = UILabel.titleLabel(text: "signin_title".localized)
+    
+    private lazy var emailLabel = UILabel.textFieldLabel(text: "signin_email_label".localized)
+    
+    private lazy var emailField = UITextField(
+        target: self,
+        placeholder: "signin_email_placeholder".localized
+    )
+    
+    private lazy var passwordLabel = UILabel.textFieldLabel(text: "signin_password_label".localized)
+    
+    private lazy var passwordField = UITextField(
+        target: self,
+        placeholder: "signin_password_placeholder".localized
+    )
+    
+    private lazy var signInButton = UIButton.primaryButton(
+        title: "signin_button_text".localized
+    )
+    
+    private lazy var forgotPasswordButton = UIButton.secondaryButton(
+        title: "forgot_password_button_text".localized
+    )
+    
+    private lazy var connectFacebookButton = UIButton.secondaryButton(
+        title: "connect_facebook_button_text".localized
+    )
+    
+    private lazy var lineView = UIView()
+    
+    private lazy var signUpButton = UIButton.secondaryButton(
+        title: "signup_button_text".localized
+    )
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        configureViews()
+    }
+    
+}
+
+
+private extension SignInViewController {
+    func configureViews() {
+        applyDefaultUIConfigs()
+        
+        let overlayImage = UIImage(named: "sign-in-overlay")
+        overlayImageView.contentMode = UIView.ContentMode.scaleAspectFill
+        overlayImageView.frame.size.width = view.bounds.width
+        overlayImageView.frame.size.height = 290
+        overlayImageView.image = overlayImage
+        
+        lineView.translatesAutoresizingMaskIntoConstraints = false
+        lineView.layer.borderWidth = 1.0
+        lineView.layer.borderColor = UIColor.black.cgColor
+        
+        view.addSubviews(subviews: [
+            overlayImageView,
+            titleLabel,
+            emailLabel,
+            emailField,
+            passwordLabel,
+            passwordField,
+            signInButton,
+            forgotPasswordButton,
+            connectFacebookButton,
+            lineView,
+            signUpButton
+        ])
+        
+        activateConstraints()
+    }
+    
+    func activateConstraints() {
+        [
+            titleLabel,
+            emailLabel,
+            emailField,
+            passwordLabel,
+            passwordField,
+            forgotPasswordButton,
+            connectFacebookButton
+        ].forEach {
+            $0.attachHorizontally(to: view)
+        }
+        
+        NSLayoutConstraint.activate([
+            titleLabel.topAnchor.constraint(
+                equalTo: view.safeAreaLayoutGuide.topAnchor,
+                constant: 100
```

Comment:
pls move all these numbers to meaningful constants. You can follow the same approach of UI.Button.constant but private to spcific views and view controllers.